plugin/storage/kafka/options.go

```diff
@@ -94,3 +125,34 @@ func (opt *Options) InitFromViper(v *viper.Viper) {
 func stripWhiteSpace(str string) string {
 	return strings.Replace(str, " ", "", -1)
 }
+
+// getCompressionLevel to get compression level from compression type
+func getCompressionLevel(compression, compressionLevel int) (int, error) {
+	switch sarama.CompressionCodec(compression) {
+	case sarama.CompressionGZIP:
+		if compressionLevel == 0 {
+			return 6, nil
```

Comment:
what are these magic numbers? Doesn't sarama define constants for them?

Reply:
Sarama doesn't define these constants. These numbers have been taken from Kafka. I have mentioned the link in PR.

Reply:
<img width="822" alt="Screenshot 2019-07-31 at 1 25 29 PM" src="https://user-images.githubusercontent.com/3134439/62194062-b1257880-b396-11e9-9c05-3cef5820725c.png">

Reply:
please include the link in the comments. And this code can be largely simplified by using a map[string]struct I suggested above.